cartography/intel/aws/ec2.py

```diff
@@ -606,3 +612,259 @@ def sync_load_balancers(session, boto3_session, regions, current_aws_account_id,
         data = get_loadbalancer_data(boto3_session, region)
         load_load_balancers(session, data, region, current_aws_account_id, aws_update_tag)
     cleanup_load_balancers(session, common_job_parameters)
+
+
+def get_ec2_vpcs(session):
+    client = session.client('ec2', config=_get_botocore_config())
+    # paginator not supported by boto
+    return client.describe_vpcs()
+
+
+def load_ec2_vpcs(session, data, current_aws_account_id, aws_update_tag):
+    # https://github.com/lyft/cartography/graphs/traffic
+    # {
+    #     "Vpcs": [
+    #         {
+    #             "VpcId": "vpc-a01106c2",
+    #             "InstanceTenancy": "default",
+    #             "Tags": [
+    #                 {
+    #                     "Value": "MyVPC",
+    #                     "Key": "Name"
+    #                 }
+    #             ],
+    #             "CidrBlockAssociations": [
+    #                 {
+    #                     "AssociationId": "vpc-cidr-assoc-a26a41ca",
+    #                     "CidrBlock": "10.0.0.0/16",
+    #                     "CidrBlockState": {
+    #                         "State": "associated"
+    #                     }
+    #                 }
+    #             ],
+    #             "State": "available",
+    #             "DhcpOptionsId": "dopt-7a8b9c2d",
+    #             "CidrBlock": "10.0.0.0/16",
+    #             "IsDefault": false
+    #         }
+    #     ]
+    # }
+
+    ingest_vpc = """
+    MERGE (new_vpc:AWSVpc{id: {VpcId}})
+    ON CREATE SET new_vpc.firstseen = timestamp(), new_vpc.vpcid ={VpcId}
+    SET new_vpc.instance_tenancy = {InstanceTenancy},
+    new_vpc.state = {State},
+    new_vpc.is_default = {IsDefault},
+    new_vpc.primary_cidr_block = {PrimaryCIDRBlock},
+    new_vpc.dhcp_options_id = {DhcpOptionsId},
+    new_vpc.lastupdated = {aws_update_tag}
+    WITH new_vpc
+    MATCH (awsAccount:AWSAccount{id: {AWS_ACCOUNT_ID}})
+    MERGE (awsAccount)-[r:RESOURCE]->(new_vpc)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = {aws_update_tag}"""
+
+    for vpc in data['Vpcs']:
+        vpc_id = vpc["VpcId"]  # fail if not present
+
+        session.run(
+            ingest_vpc,
+            VpcId=vpc_id,
+            InstanceTenancy=vpc.get("InstanceTenancy", ""),
```

Comment:
This may add complexity for queries as the attribute will not be present on the node/edge and requires the use of EXISTS(attrib) to avoid query to break. "" was used across AWS modules elsewhere

Reply:
Agreed on None